src/gpl/test/CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+find_program (BASH_PROGRAM bash)
+
+function(gpl_test test_name)
+  add_test (
+    NAME gpl.${test_name} 
+    COMMAND ${BASH_PROGRAM} ${CMAKE_CURRENT_SOURCE_DIR}/regression ${test_name}
+  )
+  set_tests_properties(gpl.${test_name}  PROPERTIES LABELS "IntegrationTest")
+endfunction()
+
+set(TEST_NAMES
```

Comment:
If this PR is merged, will these test runs twice or not at all in the current GH action?

Reply:
Not at all, tests labeled IntegrationTest are not run in the github action

Reply:
So no tests are run by this change by default?

Reply:
Correct, no change to the CI. The only change is that if you run `ctest` manually these integration tests will now run by default.

They can be excluded with the `-LE` flag as mentioned above, but the default `ninja build_and_test` rule excludes them by default which means they won't run in the CI.

Reply:
Is the goal to eventually convert over? Is the diffs file overwritten when these run in parallel?

Reply:
The goal is eventually to convert over. The diffs are overwritten in parallel mode. We probably want to switch over to writing them to stdout instead of a file.